packages/mongodb-chatbot-ui/src/MessageRating.tsx

```diff
@@ -0,0 +1,160 @@
+import { Body } from "@leafygreen-ui/typography";
+import {
+  MessageRating,
+  type MessageRatingProps,
+} from "@lg-chat/message-rating";
+import { useRef, useState } from "react";
+import { InlineMessageFeedback } from "@lg-chat/message-feedback";
+// @ts-expect-error Typescript imports of icons currently not supported
+import WarningIcon from "@leafygreen-ui/icon/dist/Warning";
+import { spacing } from "@leafygreen-ui/tokens";
+import { palette } from "@leafygreen-ui/palette";
+import { useDarkMode } from "@leafygreen-ui/leafygreen-provider";
+import { CharacterCount } from "./InputBar";
+
+export type RatingCommentStatus = "none" | "submitted" | "abandoned";
+
+export type MessageRatingWithFeedbackCommentProps = {
+  submit: (commentText: string) => void | Promise<void>;
+  abandon: () => void;
+  status: RatingCommentStatus;
+  errorMessage?: string;
+  clearErrorMessage?: () => void;
+  maxCommentCharacterCount?: number;
+  messageRatingProps: MessageRatingProps;
+};
+
+export function MessageRatingWithFeedbackComment(
+  props: MessageRatingWithFeedbackCommentProps
+) {
+  const {
+    submit,
+    abandon,
+    status,
+    errorMessage,
+    maxCommentCharacterCount,
+    messageRatingProps,
+  } = props;
+
+  const hasRating = messageRatingProps.value !== undefined;
+
+  // TODO: Use this to animate the transition after https://jira.mongodb.org/browse/LG-3965 is merged.
+  // const ratingCommentInputVisible = hasRating && status !== "submitted" && status !== "abandoned";
+
+  const ratingCommentRef = useRef(null);
+
+  const isSubmitted = status === "submitted";
+
+  const [characterCount, setCharacterCount] = useState(0);
+
+  const characterCountExceeded = maxCommentCharacterCount
+    ? characterCount > maxCommentCharacterCount
+    : false;
+
+  return (
+    <div
+      // TODO: This css is a hacky fix until https://jira.mongodb.org/browse/LG-3965 is merged
+      // Once merged we can apply this margin directly to InlineMesssageFeedback
+      className={css`
+        & > div + div {
+          margin-top: 0.5rem;
+        }
+      `}
+    >
+      <MessageRating {...messageRatingProps} />
+      {hasRating && status !== "abandoned" ? (
+        <>
+          <InlineMessageFeedback
+            ref={ratingCommentRef}
+            cancelButtonText="Cancel"
+            onCancel={() => abandon()}
+            submitButtonText="Submit"
+            onSubmit={async (e) => {
+              const form = e.target as HTMLFormElement;
+              const textarea = form.querySelector("textarea");
+              if (!characterCountExceeded) {
+                await submit(textarea?.value ?? "");
+              }
+            }}
+            textareaProps={{
+              onChange: (e) => {
+                const textarea = e.target as HTMLTextAreaElement;
+                setCharacterCount(textarea.value.length);
+              },
+              // @ts-expect-error Hacky fix for https://jira.mongodb.org/browse/LG-3964
```

Comment:
should we have separate ticket for this and the following changes that are dependent on these LG tickets?